### PR TITLE
adds a contravariant functor to spec

### PR DESCRIPTION
**Pull request by rjmk**
See https://github.com/fantasyland/fantasy-land/pull/150

Resolves https://github.com/rpominov/static-land/issues/15

### CODE REVIEW

**Comment by rpominov** [comment]
Thanks! Let's see how fantasyland/fantasy-land#150 go.

**Comment by rjmk** [comment]
👍

**Comment by rjmk** [comment]
Still interested in this now it's merged in FL? If so, I'll spend some time getting it up to date

**Comment by rpominov** [comment]
Yea, definitely interested. Please do!

**Comment by rjmk** [comment]
I appear to have stolen your profile picture, @rpominov. Sorry about that

I think this is updated! No diagram (my arch nemesis in FL PRs)

**Comment by rpominov** [comment]
Thank you! I'll take a close look tomorrow and hopefully merge.

 > I appear to have stolen your profile picture, @rpominov. Sorry about that

Haha, not quite the same though 😅

File: docs/spec.md [hunk 1]
```diff
@@ -226,6 +227,18 @@ to that of the derivation (or derivations).
 
 
 
+## Contravariant
+
+#### Methods
+
+  1. `contramap :: Contravariant f => (a → b, f b) → f a`
```
Comment (rpominov) [inline docs/spec.md]: It should be `contramap :: Contravariant f => Type f ~> (a → b, f b) → f a`.

This notations probably doesn't make much sense to anyone but me, and I [don't like it too much either](https://github.com/rpominov/static-land/issues/34). But for now we should use it in all methods for consistency.

Comment (rjmk) [inline docs/spec.md]: ⚡️

**Comment by rpominov** [comment]
Thank you!

**Pull request merged**